### PR TITLE
Refactor private constructor checker into a utility class

### PR DESCRIPTION
**Pull request by akarnokd**
Move into `TestUtil.checkUtilityClass()`.

### CODE REVIEW

**Comment by vanniktech** [comment]
👍

File: src/test/java/rx/TestUtil.java [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+ */
+
+package rx;
+
+import com.pushtorefresh.private_constructor_checker.PrivateConstructorChecker;
+
+/**
+ * Common test utility methods.
+ */
+public enum TestUtil {
```
Comment (egor-n) [inline src/test/java/rx/TestUtil.java:24]: Just curious - why use an enum here instead of a class? Is it just to forbid inheritance?

Comment (akarnokd) [inline src/test/java/rx/TestUtil.java:24]: Exactly - but not my invention. I saw this in the JDK source and in fact, saves you testing a private constructor. However, we can't expose such enums through the public API so only some internal tools are defined this way.

**Comment by davidmoten** [comment]
:+1:

**Pull request merged**